mbtiles/src/diff.rs

```diff
@@ -0,0 +1,29 @@
+use std::{path::PathBuf, str::FromStr};
+
+use crate::{Mbtiles, MbtError, MbtResult, is_empty_database};
+
+impl Mbtiles {
+    pub async fn diff(&self, dst_path: PathBuf, diff_path: PathBuf) -> MbtResult<()>{
+        let src_path = PathBuf::from_str(self.filepath()).unwrap();
+
+        if src_path == dst_path {
+            return Err(MbtError::SameSourceAndDestination(dst_path.clone()));
+        }
+        if src_path == diff_path || dst_path == diff_path {
+            return Err(MbtError::SameDiffAndSourceOrDestination(diff_path.clone()));
+        }
+
+        let dst_mbt = Mbtiles::new(&dst_path)?;
+
+        let mut dst_conn = dst_mbt.open_or_new().await?;
+
+        let is_empty_db = is_empty_database(&mut dst_conn).await?;
+        if !is_empty_db {
+            return Err(MbtError::NonEmptyTargetFile(dst_path.clone()));
```

Comment:
it's ok to have a non-empty destination - you simply append to it, just like we do with copying (assuming on-duplicates is set)